rust/ql/lib/ext/generated/rust/lang-core.model.yml

```diff
@@ -0,0 +1,2482 @@
+      - ["lang:core", "<crate::result::Result>::map_or_else", "Argument[self].Field[crate::result::Result::Ok(0)]", "Argument[1].Parameter[0]", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::ok", "Argument[self].Field[crate::result::Result::Ok(0)]", "ReturnValue.Field[crate::option::Option::Some(0)]", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::or", "Argument[0]", "ReturnValue", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::or", "Argument[self].Field[crate::result::Result::Ok(0)]", "ReturnValue.Field[crate::result::Result::Ok(0)]", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::or_else", "Argument[0].ReturnValue", "ReturnValue", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::or_else", "Argument[self].Field[crate::result::Result::Err(0)]", "Argument[0].Parameter[0]", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::or_else", "Argument[self].Field[crate::result::Result::Ok(0)]", "ReturnValue.Field[crate::result::Result::Ok(0)]", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::transpose", "Argument[self].Field[crate::result::Result::Err(0)]", "ReturnValue.Field[crate::option::Option::Some(0)].Field[crate::result::Result::Err(0)]", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::transpose", "Argument[self].Field[crate::result::Result::Ok(0)].Field[crate::option::Option::Some(0)]", "ReturnValue.Field[crate::option::Option::Some(0)].Field[crate::result::Result::Ok(0)]", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::unwrap", "Argument[self].Field[crate::result::Result::Ok(0)]", "ReturnValue", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::unwrap_err", "Argument[self].Field[crate::result::Result::Err(0)]", "ReturnValue", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::unwrap_err_unchecked", "Argument[self].Field[crate::result::Result::Err(0)]", "ReturnValue", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::unwrap_or", "Argument[0]", "ReturnValue", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::unwrap_or", "Argument[self].Field[crate::result::Result::Ok(0)]", "ReturnValue", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::unwrap_or_default", "Argument[self].Field[crate::result::Result::Ok(0)]", "ReturnValue", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::unwrap_or_else", "Argument[0].ReturnValue", "ReturnValue", "value", "dfc-generated"]
+      - ["lang:core", "<crate::result::Result>::unwrap_or_else", "Argument[self].Field[crate::result::Result::Err(0)].Reference", "ReturnValue", "value", "dfc-generated"]
```

Comment:
I don't see why this is true (the described method is [here](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or_else)).  Though (assuming I'm right) I doubt the model will do much harm anyway.

Reply:
Nicely spotted! That model is indeed odd. Both because the error value is not directly returned and because there are no references involved. The latter might be due to some mistakenly inserted reference read step.

In any case, the implementation is very simple, so I would expect the model to be accurate. I've created an internal issue for me to fix this.